pkg/api/handler.go

```diff
@@ -173,3 +176,41 @@ func (h *handler) resetPassword(c echo.Context) error {
 		"success": true,
 	})
 }
+
+func (h *handler) signinWithProvider(c echo.Context) error {
+	payload := new(oauth.Oauth2Payload)
+
+	if err := c.Bind(payload); err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, "An error occurred with your payload")
+	}
+	provider := c.Param("provider")
+	var p oauth.Provider
+	switch provider {
+	case "google":
+		p = google.New()
+	case "facebook":
+		p = facebook.New()
+	default:
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("%s provider is not supported", provider))
+	}
+	user, err := p.GetUserInfo(payload, h.config.OAuth2.State)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
+	}
+	if err := user.CreateRandomPassword(12); err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("An error occurred while creating a random password %s", err.Error()))
```

Comment:
Can you please change with `An error occurred while creating your account.` ?

Reply:
hope it's last change with this PR 😅